quiz/quiz.go

```diff
@@ -0,0 +1,179 @@
+/*
+	quiz defines a quiz and its questions and provides methods for starting and processing quizzes as well as reading in
+    a Quiz from file.
+*/
+package quiz
+
+import (
+	"bufio"
+	"encoding/csv"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"text/tabwriter"
+	"time"
+)
+
+// A Quiz struct is a collection of up to 100 Questions, the name of the file containing the quiz, the number of
+// questions in the quiz and the number of questions the user got correct.
+type Quiz struct {
+	Questions        []Question
+	Filename         string
+	TotalQuestions   int
+	CorrectQuestions int
+}
+
+// NewQuiz creates a new Quiz struct based on the information in the file at the given file name
+func NewQuiz(filename string) (*Quiz, error) {
+	fmt.Println("\nGetting quiz using file:\t", filename)
+
+	// open the file and save Question to a Quiz struct
+	csvFile, err := os.Open(filename)
+
+	if err != nil {
+		fmt.Println("Error opening CSV file:\t", err)
+		return nil, err
+	}
+
+	defer func(){
+		if csvErr := csvFile.Close(); csvErr != nil {
+			fmt.Println(csvErr)
+		}
+	}()
+
+	reader := csv.NewReader(bufio.NewReader(csvFile))
+
+	questions := make([]Question, 0, 100)
+	// loop until end of file or error is reached
+	for {
+		record, ioErr := reader.Read()
+		if ioErr == io.EOF {
+			break
+		}
+		if ioErr != nil {
+			fmt.Println("Error reading from file:\t", ioErr)
+			return nil, ioErr
+		}
+
+		questions = append(questions, Question{
+			QuestionText: record[0],
+			Answer:       record[1],
+		})
+	}
+
+	q := Quiz{
+		Questions:        questions,
+		Filename:         filename,
+		TotalQuestions:   len(questions),
+		CorrectQuestions: 0,
+	}
+
+	return &q, err
+}
+
+// Print prints a quiz to stdout with nicely aligned columns thanks to the tabWriter package. Useful for debugging.
+func (q Quiz) Print() (err error) {
+	fmt.Println("\nQuiz - ", q.Filename)
+	w := new(tabwriter.Writer)
+	w.Init(os.Stdout, 0, 8, 2, '\t', tabwriter.AlignRight)
+	_, err = fmt.Fprintln(w, "Question\tAnswer\tUser Answer\tCorrect")
+	if err != nil {
+		return err
+	}
+
+	for index, value := range q.Questions {
+		quest := q.Questions[index]
+		if quest.QuestionText != "" {
+			_, err = fmt.Fprintln(w, value)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	_, err = fmt.Fprintln(w)
+	if err != nil {
+		return err
+	}
+
+	err = w.Flush()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Start begins the user-input portion of the quiz. Each Question is presented to the user and they enter answers via
+// stdin with "\n" being used to specify the end of an answer.
+func (q *Quiz) Start(maxQuizTime int) error {
```

Comment:
```suggestion
func (q *Quiz) Start(maxQuizTime time.Duration) error {
```
These suggestions are continent for me so write up a small change but are generally not ready to be committed from github.